viewer/packages/360-images/src/icons/IconCollection.ts

```diff
@@ -154,28 +168,48 @@ export class IconCollection {
     points: Vector3[],
     sceneHandler: SceneHandler,
     onBeforeSceneRendered: EventTrigger<BeforeSceneRenderedDelegate>
-  ): Image360Icon[] {
-    return points.map(
-      point =>
-        new Image360Icon(
-          point,
-          this._hoverIconTexture,
-          sceneHandler,
-          this.MIN_PIXEL_SIZE,
-          this.MAX_PIXEL_SIZE,
-          this._iconRadius,
-          onBeforeSceneRendered
-        )
+  ): Overlay3DIcon[] {
+    sceneHandler.addCustomObject(this._hoverSprite);
+
+    const icons = points.map(point => {
+      const icon = new Overlay3DIcon({
+        position: point,
+        minPixelSize: this.MIN_PIXEL_SIZE,
+        maxPixelSize: this.MAX_PIXEL_SIZE,
+        iconRadius: this._iconRadius,
+        hoverSprite: this._hoverSprite
+      });
+
+      return icon;
+    });
+
+    const renderSize = new Vector2();
+
+    onBeforeSceneRendered.subscribe(({ renderer, camera }) =>
+      icons.forEach(icon =>
+        icon.updateAdaptiveScale({ camera, renderSize: renderer.getSize(renderSize), domElement: renderer.domElement })
+      )
     );
+
+    return icons;
   }
 
   public dispose(): void {
     this._onBeforeSceneRenderedEvent.unsubscribe(this._activeCullingSchemeEventHandeler);
-    this._sceneHandler.removeCustomObject(this._iconsSprite);
-    this._iconsSprite.dispose();
+    this._sceneHandler.removeCustomObject(this._pointsObject);
+    this._pointsObject.dispose();
     this._sharedTexture.dispose();
   }
 
+  private createHoverSprite(hoverIconTexture: THREE.CanvasTexture): THREE.Sprite {
+    const spriteMaterial = new SpriteMaterial({ map: hoverIconTexture, depthTest: false });
+    const sprite = new Sprite(spriteMaterial);
+    sprite.updateMatrixWorld();
+    sprite.visible = false;
+    sprite.renderOrder = 5;
```

Comment:
We should find a better way of defining render order than hard-coding them like this :sweat_smile: But Let's do it in another PR. I need this for annotations as well